projects/igniteui-angular/src/lib/data-operations/sorting-strategy.ts

```diff
@@ -163,26 +164,34 @@ export class IgxSorting implements IGridSortingStrategy {
         let resolvedValue = resolveNestedPath(obj, key);
         if (isDate || isTime) {
             const date = parseDate(resolvedValue);
-            resolvedValue  = isTime && date ?
+            resolvedValue = isTime && date ?
                 new Date().setHours(date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds()) : date;
 
         }
         return resolvedValue;
     }
 
     private groupedRecordsByExpression(data: any[],
-            index: number,
-            expression: IGroupingExpression,
-            isDate: boolean = false): any[] {
+        index: number,
+        expression: IGroupingExpression,
+        isDate: boolean = false,
+        grid?: GridType,
+        groupRow?: IGroupByRecord): any[] {
         const res = [];
         const key = expression.fieldName;
         const len = data.length;
         const groupval = this.getFieldValue(data[index], key, isDate);
         res.push(data[index]);
+        if (grid && groupRow) {
+            grid.selectionService.rowsDirectParents.set(grid.selectionService.getRowID(data[index]), groupRow);
```

Comment:
Actually, since this is a sorting strategy, as the name implies it's user-configurable which means you shouldn't even rely on it to do anything extra outside what the `ISortingStrategy` describes since it can be replaced. That makes this bit of the approach a no-go.

Besides that, I've looked at the performance for this scenario and there's a much easier low-hanging fruit that will reduce the delay significantly (and thus the caching benefit). This can still be useful, but it's benefits will be noticeable only in much larger data sets.